src/Api/Controllers/AppsController.php

```diff
@@ -43,63 +40,31 @@ public function onConstruct()
         $this->model = new Apps();
         $this->additionalSearchFields = [
             ['is_deleted', ':', '0'],
+            ['id', ':', implode('|', $this->userData->getAssociatedApps())],
         ];
     }
 
     /**
-     * List of data.
+     * Format output.
      *
-     * @method GET
-     * url /v1/data
-     *
-     * @param int $id
-     * @return \Phalcon\Http\Response
+     * @param [type] $results
+     * @return void
      */
-    public function index($id = null): Response
+    protected function processOutput($results)
     {
-        if ($id != null) {
-            return $this->getById($id);
-        }
-
-        //parse the rquest
-        $parse = new QueryParserCustomFields($this->request->getQuery(), $this->model);
-        $parse->appendParams($this->additionalSearchFields);
-        $parse->appendCustomParams($this->additionalCustomSearchFields);
-        $parse->appendRelationParams($this->additionalRelationSearchFields);
-        $params = $parse->request();
-
-        $results = (new SimpleRecords(null, $this->model, $this->model->getReadConnection()->query($params['sql'], $params['bind'])));
-        $count = $this->model->getReadConnection()->query($params['countSql'], $params['bind'])->fetch(\PDO::FETCH_OBJ)->total;
-
-        // Relationships, but we have to change it to sparo full implementation
-        if ($this->request->hasQuery('relationships')) {
-            $relationships = $this->request->getQuery('relationships', 'string');
-
-            $results = QueryParser::parseRelationShips($relationships, $results);
-        }
-
-
-
-        $newResult = $this->mapper->mapMultiple(iterator_to_array($results), DTOAppsSettings::class);
-
-        //this means the want the response in a vuejs format
-        if ($this->request->hasQuery('format')) {
-            $limit = (int) $this->request->getQuery('limit', 'int', 25);
-
-            $newResult = [
-                'data' => $results,
-                'limit' => $limit,
-                'page' => $this->request->getQuery('page', 'int', 1),
-                'total_pages' => ceil($count / $limit),
-                'total_rows' => $count
-            ];
-        }
-
-        return $this->response($newResult);
+        //DTOAppsSettings
+        $this->dtoConfig->registerMapping(Apps::class, AppsSettings::class)
+          ->forMember('settings', function (Apps $app) {
+              return $app->settingsApp->toArray();
+          });
+
+        return $results instanceof \Phalcon\Mvc\Model\Resultset\Simple ?
+                $this->mapper->mapMultiple(iterator_to_array($results), AppsSettings::class)
```

Comment:
Since Mapper Plus 1.3.5 all iterable structures can be passed to mapMultiple.
So, Simple object can be passed directly to mapMultiple.